lib/ts/recipe/usermetadata/types.ts

```diff
@@ -0,0 +1,75 @@
+/* Copyright (c) 2022, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import OverrideableBuilder from "supertokens-js-override";
+
+export type JSONPrimitive = string | number | boolean | null;
+export type JSONArray = Array<JSONPrimitive>;
+export type JSONValue = JSONPrimitive | JSONObject | JSONArray;
+export interface JSONObject {
+    [ind: string]: JSONValue;
+}
+
+export type TypeInput = {
+    override?: {
+        functions?: (
```

Comment:
I don't think an `apis` prop would be good here: it can only be empty and this way it's even more clear that this recipe provides no APIs. On the other hand, all other recipes have it...

Reply:
We should have it, and it can be empty.